interpreter/go/go.go

```diff
@@ -0,0 +1,170 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package golang // import "go.opentelemetry.io/ebpf-profiler/interpreter/go"
+
+/*
+#cgo CFLAGS: -g -Wall
+#include "../../rust-crates/symblib-capi/c/symblib.h"
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"runtime"
+	"sync/atomic"
+	"unsafe"
+
+	"go.opentelemetry.io/ebpf-profiler/host"
+	"go.opentelemetry.io/ebpf-profiler/interpreter"
+	"go.opentelemetry.io/ebpf-profiler/libpf"
+	"go.opentelemetry.io/ebpf-profiler/metrics"
+	"go.opentelemetry.io/ebpf-profiler/remotememory"
+	"go.opentelemetry.io/ebpf-profiler/reporter"
+	"go.opentelemetry.io/ebpf-profiler/successfailurecounter"
+)
+
+var (
+	// compiler check to make sure the needed interfaces are satisfied
+	_ interpreter.Data     = &goData{}
+	_ interpreter.Instance = &goInstance{}
+)
+
+type goData struct {
+	exec string
+}
+
+type goInstance struct {
+	interpreter.InstanceStubs
+
+	// Go symbolization metrics
+	successCount atomic.Uint64
+	failCount    atomic.Uint64
+
+	goRuntime *C.SymblibPointResolver
+	pin       runtime.Pinner
+}
+
+func Loader(_ interpreter.EbpfHandler, info *interpreter.LoaderInfo) (
+	interpreter.Data, error) {
+	ef, err := info.GetELF()
+	if err != nil {
+		return nil, err
+	}
+	if !ef.IsGolang() {
+		return nil, nil
+	}
+
+	exec := info.FileName()
+	if len(exec) < 2 {
+		// There are cases where FileName() just returns /.
+		// In these cases we can not access the backing executable
+		// by FileName() and therefore can not continue here.
+		return nil, nil
+	}
+
+	return &goData{
+		exec: exec,
+	}, nil
+}
+
+func (g *goData) Attach(_ interpreter.EbpfHandler, pid libpf.PID,
+	_ libpf.Address, _ remotememory.RemoteMemory) (interpreter.Instance, error) {
+	gi := &goInstance{}
+
+	executablePath := C.CString(fmt.Sprintf("/proc/%d/root%s", pid, g.exec))
```

Comment:
We should use `filepath.Join` here instead of hacking the concatenation together, it handles absolute and relative paths and makes no assumptions:

```suggestion
	executablePath := C.CString(filepath.Join(fmt.Sprintf("/proc/%d/root", pid), g.exec))
```